### ISSUE TITLE
ValueError and BrokenPipeError in max_storage_temp_tutorial notebook

### ISSUE THREAD

**Issue by clayms** [HazyResearch/fonduer-tutorials#12]
When running the following code cell from max_storage_temp_tutorial.ipynb :

```
corpus_parser = Parser(structural=True, lingual=True, visual=True, pdf_path=pdf_path)
%time corpus_parser.apply(doc_preprocessor, parallelism=PARALLEL)
```

I get the following error repeated 4 times where the only thing that changes is the number after the `UDF-` :

```
Process ParserUDF-8:
Traceback (most recent call last):
  File "/home/user/anaconda3/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/home/clay.stevens/anaconda3/lib/python3.6/site-packages/fonduer/utils/udf.py", line 169, in run
    for y in self.apply(x, **self.apply_kwargs):
  File "/home/user/anaconda3/lib/python3.6/site-packages/fonduer/parser/parser.py", line 143, in apply
    document.name, document.sentences, self.pdf_path
  File "/home/user/anaconda3/lib/python3.6/site-packages/fonduer/parser/visual_linker.py", line 38, in parse_visual
    self.extract_pdf_words()
  File "/home/user/anaconda3/lib/python3.6/site-packages/fonduer/parser/visual_linker.py", line 54, in extract_pdf_words
    for i in range(1, int(num_pages) + 1):
ValueError: invalid literal for int() with base 10: b''
```

Then the following plain text is printed:

```
CPU times: user 8.49 s, sys: 121 ms, total: 8.61 s
Wall time: 44 s
```

Then the following error repeats 96 times:

```
Traceback (most recent call last):
  File "/home/user/anaconda3/lib/python3.6/multiprocessing/queues.py", line 240, in _feed
    send_bytes(obj)
  File "/home/user/anaconda3/lib/python3.6/multiprocessing/connection.py", line 200, in send_bytes
    self._send_bytes(m[offset:offset + size])
  File "/home/user/anaconda3/lib/python3.6/multiprocessing/connection.py", line 397, in _send_bytes
    self._send(header)
  File "/home/user/anaconda3/lib/python3.6/multiprocessing/connection.py", line 368, in _send
    n = write(self._handle, buf)
BrokenPipeError: [Errno 32] Broken pipe
```

**Comment by clayms** [comment]
I set `parallelism=None` which resulting in the following single error:

```
ValueError                                Traceback (most recent call last)
<timed eval> in <module>()

~/anaconda3/lib/python3.6/site-packages/fonduer/utils/udf.py in apply(self, xs, clear, parallelism, progress_bar, count, **kwargs)
     48         self.logger.info("Running UDF...")
     49         if parallelism is None or parallelism < 2:
---> 50             self.apply_st(xs, progress_bar, clear=clear, count=count, **kwargs)
     51         else:
     52             self.apply_mt(xs, parallelism, clear=clear, **kwargs)

~/anaconda3/lib/python3.6/site-packages/fonduer/utils/udf.py in apply_st(self, xs, progress_bar, count, **kwargs)
     71 
     72             # Apply UDF and add results to the session
---> 73             for y in udf.apply(x, **kwargs):
     74 
     75                 # If UDF has a reduce step, this will take care of the insert;

~/anaconda3/lib/python3.6/site-packages/fonduer/parser/parser.py in apply(self, x, **kwargs)
    141                 logger.error("Visual parsing failed: pdf files are required")
    142             yield from self.vizlink.parse_visual(
--> 143                 document.name, document.sentences, self.pdf_path
    144             )
    145         else:

~/anaconda3/lib/python3.6/site-packages/fonduer/parser/visual_linker.py in parse_visual(self, document_name, sentences, pdf_path)
     36             self.pdf_file = self.pdf_file[:-3] + "PDF"
     37         try:
---> 38             self.extract_pdf_words()
     39         except RuntimeError as e:
     40             self.logger.exception(e)

~/anaconda3/lib/python3.6/site-packages/fonduer/parser/visual_linker.py in extract_pdf_words(self)
     52         pdf_word_list = []
     53         coordinate_map = {}
---> 54         for i in range(1, int(num_pages) + 1):
     55             html_content = subprocess.check_output(
     56                 "pdftotext -f {} -l {} -bbox-layout '{}' -".format(

ValueError: invalid literal for int() with base 10: b''
```

**Comment by lukehsiao** [comment]
Which version of poppler-utils are you using?

It appears like `num_pages` isn't being calculated correctly. Is your input document a PDF? Are you able to narrow the error down to a single input document that you could attach?

Note that if your input documents are purely HTML and not PDF, you'll need to set `visual=False` in the parser.

**Comment by clayms** [comment]
poppler-0:0.26.5-17.el7_4.x86_64

Currently I am waiting on Admin to update poppler on this machine.  In the mean time I switched to a machine where I have Admin rights.

The pdfs I am using are the ones downloaded from `hardware/download_data.sh`.

With the latest version of poppler, this error went away.
However, now I am getting the following repeating error, which I will report as a separate issue and close this issue.

```
UnicodeEncodeError: 'ascii' codec can't encode character '\uf0b7' in position 8479: ordinal not in range(128)
```

**Comment by oleg1187** [comment]
I am having a similar issue with the parser in the transistor sheet tutorial. Can anyone help?

![image](https://user-images.githubusercontent.com/31730615/49256665-07b96c00-f430-11e8-886b-89753b58356a.png)
![image](https://user-images.githubusercontent.com/31730615/49256687-21f34a00-f430-11e8-97a1-a017dacb6c8b.png)

**Comment by lukehsiao** [comment]
Can you open a new issue? When you do, please include which version of poppler you have and a sample document exhibiting the behavior.
